walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,66 +27,74 @@
         Collection<TimeRange> avaliableTimes = new ArrayList<>();
         long duration = request.getDuration();
         Collection<String> attendees = request.getAttendees();
+        Collection<String> optionalAttendees = request.getOptionalAttendees();
 
         // Edge Cases
         if (duration >= TimeRange.WHOLE_DAY.duration()) {
             return avaliableTimes;
         }
-        else if (attendees.isEmpty() || events.isEmpty()){
+        else if ((attendees.isEmpty() && optionalAttendees.isEmpty()) || events.isEmpty()){
             avaliableTimes.add(TimeRange.WHOLE_DAY);
             return avaliableTimes;
         }
 
         // Get Collection of unavaliableTimes times
         Collection<TimeRange> unavaliableTimes = getUnavaliableTimes(events, attendees);
         // Sort Unavaliable times by start time
-        List<TimeRange> unavTimesList = unavaliableTimes.stream().sorted(TimeRange.ORDER_BY_START).collect(Collectors.toList());
+        List<TimeRange> unavaliableTimesList = unavaliableTimes.stream().sorted(TimeRange.ORDER_BY_START).collect(Collectors.toList());
         // Create Avaliable times
-        avaliableTimes = generateAvaliableTimes(unavTimesList, duration);
+        avaliableTimes = generateAvaliableTimes(unavaliableTimesList, duration);
+
+        // If no conflicting times for attendees, check optional attendees
+        if (avaliableTimes.equals(TimeRange.WHOLE_DAY)){
+            unavaliableTimes = getUnavaliableTimes(events, optionalAttendees);
+            unavaliableTimesList = unavaliableTimes.stream().sorted(TimeRange.ORDER_BY_START).collect(Collectors.toList());
```

Comment:
+1

The reason for this is so that you code base stays consistent. I am fine with all streams or all for loops, but having both is just not stylistic